Apps/phfireeyeetp/fireeyeetp_connector.py

```diff
@@ -0,0 +1,1491 @@
+        # the error and adds it to the action_result.
+        if 'html' in r.headers.get('Content-Type', ''):
+            return self._process_html_response(r, action_result)
+
+        # it's not content-type that is to be parsed, handle an empty response
+        if not r.text:
+            return self._process_empty_response(r, action_result)
+
+        # everything else is actually an error at this point
+        message = "Can't process response from server. Status Code: {0} Data from server: {1}".format(
+            r.status_code,
+            self._handle_py_ver_compat_for_input_str(r.text.replace('{', '{{').replace('}', '}}'))
+        )
+
+        return RetVal(action_result.set_status(phantom.APP_ERROR, message), None)
+
+    def _make_rest_call(self, endpoint, action_result, method="get", **kwargs):
+        # **kwargs can be any additional parameters that requests.request accepts
+
+        config = self.get_config()
+
+        resp_json = None
+
+        try:
+            request_func = getattr(requests, method)
+        except AttributeError:
+            return RetVal(
+                action_result.set_status(phantom.APP_ERROR, "Invalid method: {0}".format(method)),
+                resp_json
+            )
+
+        try:
+            # Create a URL to connect to
+            url = "{}{}".format(self._base_url, endpoint)
+        except Exception:
+            err = "Please provide valid asset configuration and|or the action parameters"
+            return RetVal(action_result.set_status(phantom.APP_ERROR, err), resp_json)
+        try:
+            r = request_func(
+                url,
+                verify=config.get('verify_server_cert', False),
+                headers=self._header,
+                **kwargs
+            )
+        except requests.exceptions.InvalidSchema:
+            err = 'Error connecting to server. No connection adapters were found for %s' % (url)
+            return RetVal(action_result.set_status(phantom.APP_ERROR, err), resp_json)
+        except requests.exceptions.InvalidURL:
+            err = 'Error connecting to server. Invalid URL %s' % (url)
+            return RetVal(action_result.set_status(phantom.APP_ERROR, err), resp_json)
+        except Exception as e:
+            err = self._get_error_message_from_exception(e)
+            return RetVal(
+                action_result.set_status(
+                    phantom.APP_ERROR, "Error Connecting to server. Details: {0}".format(err)
+                ), resp_json
+            )
+
+        return self._process_response(r, action_result)
+
+    def _save_file_to_vault(self, data, filename, container_id, action_result):
+        # Creating temporary directory and file
+        try:
+            if hasattr(Vault, 'get_vault_tmp_dir'):
+                temp_dir = Vault.get_vault_tmp_dir()
+            else:
+                temp_dir = "/opt/phantom/vault/tmp/"
+
+            temp_dir = "{}{}".format(temp_dir, '/{}'.format(uuid.uuid4()))
```

Comment:
I think we don't need the nested format statements here
```suggestion
            temp_dir = "{}/{}".format(temp_dir, uuid.uuid4())
```

Reply:
Fixed